fedimint-load-test-tool/src/main.rs

```diff
@@ -0,0 +1,627 @@
+        initial_notes: Option<TieredMulti<SpendableNote>>,
+
+        #[arg(
+            long,
+            help = "Gateway public key. If none, retrieve one according to --generate-invoice-with"
+        )]
+        gateway_public_key: Option<String>,
+
+        #[arg(
+            long,
+            help = "Whether to use lncli or lightning-cli to generate invoices to be paid through the gateway. If none, no gateway/LN tests will be run. Note that you can't generate an invoice using same lightning node used by the gateway (i.e self payment is forbidden)"
+        )]
+        generate_invoice_with: Option<LnType>,
+    },
+}
+
+#[derive(Debug, Clone)]
+struct Event(String, Duration);
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    fedimint_logging::TracingSetup::default().init()?;
+    let opts = Opts::parse();
+    let (event_sender, mut event_receiver) = tokio::sync::mpsc::unbounded_channel();
+    let futures = match opts.command {
+        Command::TestDownload { connect } => {
+            test_download_config(&connect, opts.users, event_sender.clone()).await?
+        }
+        Command::LoadTest {
+            connect,
+            initial_notes,
+            generate_invoice_with,
+            gateway_public_key,
+        } => {
+            let connect = if let Some(connect) = connect {
+                connect.to_owned()
+            } else {
+                cmd!("fedimint-cli", "connect-info").out_json().await?["connect_info"]
+                    .as_str()
+                    .map(ToOwned::to_owned)
+                    .expect("connect-info command to succeed")
+            };
+            let initial_notes = if let Some(initial_notes) = initial_notes {
+                initial_notes.to_owned()
+            } else {
+                cmd!("fedimint-cli", "spend", "2000sats").out_json().await?["note"]
+                    .as_str()
+                    .map(parse_ecash)
+                    .transpose()?
+                    .expect("spend command to succeed")
+            };
+
+            let gateway_public_key = if let Some(gateway_public_key) = gateway_public_key {
+                Some(gateway_public_key)
+            } else if let Some(generate_invoice_with) = generate_invoice_with {
+                Some(get_gateway_public_key(generate_invoice_with).await?)
+            } else {
+                None
+            };
+            if generate_invoice_with.is_none() {
+                warn!("No --generate-invoice-with given, not LN/gateway tests will be run")
+            }
+            run_economy(
+                opts.users,
+                connect,
+                initial_notes,
+                generate_invoice_with,
+                gateway_public_key,
+                event_sender.clone(),
+            )
+            .await?
+        }
+    };
+    let summary_handle = tokio::spawn(async move {
+        let mut results = HashMap::new();
+        while let Some(event) = event_receiver.recv().await {
```

Comment:
We should probably aggregate here (eg. each time there's more than N events, aggregate to it's own list), otherwise this might quickly balloon, but good enough for time being.

Reply:
This is really far from being an issue right know